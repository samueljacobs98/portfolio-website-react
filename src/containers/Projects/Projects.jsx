import Card from "../../components/Projects/Card/Card";
import Project from "../../components/Projects/Project/Project";
import moon from "../../assets/svgs/moon.svg";
import "./Projects.scss";
import projectData from "../../assets/data/projectData";
import { useEffect, useState } from "react";
import Button from "../../components/Button/Button";

const Projects = () => {
  const [windowIsBig, setWindowIsBig] = useState(window.innerWidth > 600);
  const [projects, setProjects] = useState([...projectData]);
  const [toggle, setToggle] = useState(false);
  const [projectToShow, setProjectToShow] = useState(0);
  // console.log(projectData[0]);

  // console.log(projects);

  const handleResize = () => {
    setWindowIsBig(window.innerWidth > 600);
  };

  window.addEventListener("resize", handleResize);

  useEffect(() => {
    const projectsToShow =
      !windowIsBig && toggle ? [...projectData].slice(0, 3) : [...projectData];
    setProjects([...projectsToShow]);
  }, [windowIsBig, toggle]);

  useEffect(() => {
    setToggle(!windowIsBig);
  }, [windowIsBig]);

  useEffect(() => {
    toggle
      ? setProjects([...projectData].slice(0, 3))
      : setProjects([...projectData]);
  }, [toggle]);

  useEffect(() => {
    if (!windowIsBig && projectToShow > 2) setProjectToShow(0);
  }, [windowIsBig]);

  useEffect(() => {
    if (toggle && projectToShow > 2) setProjectToShow(0);
  }, [toggle]);

  const showMore = () => {
    setToggle(!toggle);
  };

  const handleClick = (id) => {
    setProjectToShow(id);
  };

  const handleIncrement = () => {
    projectToShow === projectData.length - 1
      ? setProjectToShow(0)
      : setProjectToShow(projectToShow + 1);
  };

  const handleDecrement = () => {
    projectToShow === 0
      ? setProjectToShow(projectData.length - 1)
      : setProjectToShow(projectToShow - 1);
  };

  const cardListJSX = (
    <>
      {projects.map((project, index) => {
        return (
          <Card
            key={project.id}
            data={project}
            handleClick={handleClick}
            index={index}
            projectToShow={projectToShow}
          />
        );
      })}
    </>
  );

  return (
    <div className="projects" id="projects">
      <Project
        data={projects[projectToShow]}
        handleIncrement={handleIncrement}
        handleDecrement={handleDecrement}
      />
      <main className="projects__container">
        <h3 className="projects__title">Featured Projects</h3>
        {cardListJSX}
        {!windowIsBig && (
          <div className="projects__button-container">
            <Button
              text={toggle ? "Show more..." : "Show less"}
              onClick={showMore}
            />
          </div>
        )}
      </main>
      <img className="projects__moon" src={moon} alt="the moon" />
    </div>
  );
};

export default Projects;
