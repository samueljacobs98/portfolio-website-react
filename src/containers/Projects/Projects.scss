@use "../../assets/scss/variables" as *;
@use "../../assets/scss/mixins";

.projects {
  @include mixins.sectionSetup;
  position: relative;
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
  gap: $gap-project;
  padding: $padding-projects;

  &__container {
    max-width: 800px;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    gap: $gap-project;
    .projects__title {
      width: 100%;
      text-align: center;
    }
  }
  &__moon {
    @include mixins.floatingAnimation;
    top: 4%;
    left: 65%;
    z-index: -1;
  }

  &__button-container {
    width: 100%;
    display: flex;
    justify-content: center;
  }
}

@media screen and (max-width: 1060px) {
  .projects {
    padding-top: 3rem;
    display: grid;
    &__container {
      order: 1;
    }
  }
}

@media screen and (max-width: 600px) {
  .projects {
    display: grid;
    &__container {
      grid-template-columns: 1fr;
      justify-items: center;
    }
  }
}
