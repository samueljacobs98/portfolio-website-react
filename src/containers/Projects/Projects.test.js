import { render } from "@testing-library/react";
import Projects from "./Projects";

test("should match snapshot", () => {
  const { container } = render(<Projects />);
  expect(container).toMatchSnapshot();
});
