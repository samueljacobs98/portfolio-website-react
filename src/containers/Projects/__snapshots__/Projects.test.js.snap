// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="projects"
    id="projects"
  >
    <section
      class="project"
      id="project"
    >
      <h3
        class="project__heading"
      >
        JAMstack and CMS Blog
      </h3>
      <img
        alt="JAMstack and CMS Blog"
        class="project__img"
        src="blogscreenshot.png"
      />
      <div
        class="project__text"
      >
        <p
          class="project__para"
        >
          The goal of this project was to create a fast and secure blog using JAMStack. Eleventy was used to create a static site, using Netlify CMS to manage adding, removing, and editing blog posts.
        </p>
        <br />
        <p
          class="project__para"
        >
          A fundamental goal of this website was to create a page that utilised standardised components as much as possible. In doing so, nunjucks and eleventy could be used to integrate multiple components on a range of pages. Standardised components include the base file, header, footer, and article-snippet.
        </p>
        <br />
        <p
          class="project__para"
        >
          Using the Netlify CMS, and by implementing an admin file, a secure admin section of the site was added, enabling any designated user to access the admin interface, thus enabling them to create new, edit existing, and delete old posts on the blog.
        </p>
      </div>
      <div
        class="project__skills"
      >
        <p
          class="project__skill"
        >
          HTML/CSS
        </p>
        <p
          class="project__skill"
        >
          Eleventy
        </p>
        <p
          class="project__skill"
        >
          Netlify CMS
        </p>
        <p
          class="project__skill"
        >
          Nunjucks
        </p>
      </div>
      <div
        class="project-links"
      >
        <a
          class="project-links__container"
          href="https://github.com/samueljacobs98/jamstack-blog"
          rel="noreferrer"
          target="_blank"
        >
          <p
            class="project-links__text"
          >
            Code:
          </p>
          <img
            alt=""
            class="project-links__link"
            src="github.svg"
          />
        </a>
        <a
          class="project-links__container"
          href="https://sage-chaja-b1e942.netlify.app/"
          rel="noreferrer"
          target="_blank"
        >
          <p
            class="project-links__text"
          >
            Preview:
          </p>
          <img
            alt=""
            class="project-links__link"
            src="preview.svg"
          />
        </a>
      </div>
      <div
        class="project__update"
      >
        <button
          class="button"
        >
          Previous Project
        </button>
        <button
          class="button"
        >
          Next Project
        </button>
      </div>
    </section>
    <main
      class="projects__container"
    >
      <h3
        class="projects__title"
      >
        Featured Projects
      </h3>
      <a
        class="card card--active"
        href="#project"
      >
        <p
          class="card__title"
        >
          JAMstack and CMS Blog
        </p>
        <p
          class="card__brief"
        >
          A fast and secure blog with Eleventy and Netlify CMS
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              HTML/CSS
            </p>
            <p
              class="skill skill2"
            >
              Eleventy
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              Netlify CMS
            </p>
            <p
              class="skill skill4"
            >
              Nunjucks
            </p>
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          GCP Hosted Pokemon API
        </p>
        <p
          class="card__brief"
        >
          Created an API using Spring Boot and Java with a MySQL database.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              Google Cloud
            </p>
            <p
              class="skill skill2"
            >
              Spring Boot
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              Java
            </p>
            <p
              class="skill skill4"
            >
              MySQL
            </p>
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          2048 Game
        </p>
        <p
          class="card__brief"
        >
          A browser-based game using modern JavaScript programming techniques and DOM manipulation.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              JavaScript
            </p>
            <p
              class="skill skill2"
            >
              HTML
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              CSS
            </p>
            <p
              class="skill skill4"
            >
              Figma
            </p>
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          General BG Animation
        </p>
        <p
          class="card__brief"
        >
          Website where users to modify the background.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              HTML Canvas
            </p>
            <p
              class="skill skill2"
            >
              JavaScript
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              HTML
            </p>
            <p
              class="skill skill4"
            >
              CSS
            </p>
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          Calculator
        </p>
        <p
          class="card__brief"
        >
          A browser-based calculator capable of basic and advanced calculations.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              HTML
            </p>
            <p
              class="skill skill2"
            >
              CSS
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              JavaScript
            </p>
            <p
              class="skill skill4"
            >
              Figma
            </p>
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          Ticket Tracker
        </p>
        <p
          class="card__brief"
        >
          A React app that displays team members' remaining tickets.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              React
            </p>
            <p
              class="skill skill2"
            >
              State-management
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              HTML Canvas
            </p>
            <p
              class="skill skill4"
            />
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          Punk API
        </p>
        <p
          class="card__brief"
        >
          A React application that pulls data from an external public API and presents it in a dashboard.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              Async Coding
            </p>
            <p
              class="skill skill2"
            >
              API Integration
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              React
            </p>
            <p
              class="skill skill4"
            />
          </div>
        </div>
      </a>
      <a
        class="card"
        href="#project"
      >
        <p
          class="card__title"
        >
          MyVocab
        </p>
        <p
          class="card__brief"
        >
          A simple DOM manipulation project using API integration.
        </p>
        <div
          class="card__line"
        />
        <div
          class="card__skills"
        >
          <div
            class="card__skills-left"
          >
            <p
              class="skill skill1"
            >
              JavaScript
            </p>
            <p
              class="skill skill2"
            >
              API Integration
            </p>
          </div>
          <div
            class="card__skills-right"
          >
            <p
              class="skill skill3"
            >
              SCSS
            </p>
            <p
              class="skill skill4"
            >
              HTML
            </p>
          </div>
        </div>
      </a>
    </main>
    <img
      alt="the moon"
      class="projects__moon"
      src="moon.svg"
    />
  </div>
</div>
`;
